internal/cssc/cssc.go

```diff
@@ -0,0 +1,190 @@
+	}
+
+	return filter, nil
+}
+
+// Applies filter to filter out the repositories and tags from the ACR according to the specified criteria and returns the FilteredRepository struct
+func ApplyFilterAndGetFilteredList(ctx context.Context, acrClient api.AcrCLIClientInterface, filter Filter) ([]FilteredRepository, error) {
+	var filteredRepos []FilteredRepository
+
+	for _, filterRepo := range filter.Repositories {
+		if filterRepo.Enabled != nil && !*filterRepo.Enabled {
+			continue
+		}
+		if filterRepo.Repository == "" || filterRepo.Tags == nil || len(filterRepo.Tags) == 0 || filterRepo.Tags[0] == "" {
+			continue
+		}
+
+		tagList, err := tag.ListTags(ctx, acrClient, filterRepo.Repository)
+		if err != nil {
+			if strings.Contains(err.Error(), "404") {
+				continue
+			}
+			return nil, err
+		}
+
+		if len(filterRepo.Tags) == 1 && filterRepo.Tags[0] == "*" {
+			for _, tag := range tagList {
+				if strings.Contains(*tag.Name, "-patched") {
+					originalTag := strings.Split(*tag.Name, "-patched")[0]
+					matchingRepo := FilteredRepository{Repository: filterRepo.Repository, Tag: originalTag, PatchTag: *tag.Name}
+					filteredRepos = AppendElement(filteredRepos, matchingRepo)
+				} else {
+					matchingRepo := FilteredRepository{Repository: filterRepo.Repository, Tag: *tag.Name, PatchTag: *tag.Name}
+					filteredRepos = AppendElement(filteredRepos, matchingRepo)
+				}
+			}
+		} else {
+			for _, ftag := range filterRepo.Tags {
+				for _, tag := range tagList {
+					if *tag.Name == ftag {
+						matchingRepo := FilteredRepository{Repository: filterRepo.Repository, Tag: *tag.Name, PatchTag: *tag.Name}
+						filteredRepos = AppendElement(filteredRepos, matchingRepo)
+					} else if *tag.Name == ftag+"-patched" {
+						matchingRepo := FilteredRepository{Repository: filterRepo.Repository, Tag: ftag, PatchTag: ftag + "-patched"}
+						filteredRepos = AppendElement(filteredRepos, matchingRepo)
+					}
+				}
+			}
+		}
+	}
+	return filteredRepos, nil
+}
+
+// Prints the filtered result to the console
+func PrintFilteredResult(filteredResult []FilteredRepository, showPatchTags bool, loginURL string) {
+	if len(filteredResult) == 0 {
+		fmt.Println("No matching repository and tag found!")
+	}
+	if showPatchTags {
+		fmt.Println("Listing repositories and tags matching the filter with corrosponding patch tag (if present):")
+		for _, result := range filteredResult {
+			fmt.Printf("%s/%s:%s,%s\n", loginURL, result.Repository, result.Tag, result.PatchTag)
+		}
+	} else {
+		fmt.Println("Listing repositories and tags matching the filter:")
+		for _, result := range filteredResult {
+			fmt.Printf("%s/%s:%s\n", loginURL, result.Repository, result.Tag)
+		}
+	}
+	fmt.Println("Total matches found:", len(filteredResult))
+}
+
+// Appends the element to the slice if it does not already exist in the slice
+func AppendElement(slice []FilteredRepository, element FilteredRepository) []FilteredRepository {
```

Comment:
This is not efficient. Use a set or I suppose in go a map[string]struct{} since go technically doesn't have sets.

Reply:
I realized I had a double loop there. Simplified it now.

Reply:
Updated the code to update the slice element if it exists. Thanks for the suggestion and help.